index.d.ts

```diff
@@ -185,9 +185,9 @@ declare module 'focus-trap' {
     delayInitialFocus?: boolean;
     /**
      * Default: `window.document`. Document where the focus trap will be active.
-     * This allows to use FocusTrap in an iFrame context.
+     * This allows to use FocusTrap in an iFrame context or to configure the focus trap to use your shadow host element as its document.
      */
-    document?: Document;
+    document?: Document | HTMLElement;
```

Comment:
The only other option would be https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot because of the expectation to use the `.activeElement` API. But I don't think this is necessary. See my comment on the related issue you opened earlier.